tests/Unit/Utilities/Test_TMPLDocumentation.cpp

```diff
@@ -0,0 +1,1467 @@
+/// [tmpl::range]
+
+/// [tmpl::reverse_range]
+assert_same<tmpl::reverse_range<size_t, 7, 4>,
+            tmpl::list<tmpl::size_t<7>, tmpl::size_t<6>, tmpl::size_t<5>>>();
+assert_same<tmpl::reverse_range<size_t, 7, 7>, tmpl::list<>>();
+/// [tmpl::reverse_range]
+}
+}  // namespace list_constructors
+
+namespace list_query {
+/// [tmpl::count_if:bug:definitions]
+struct TrueFalse {
+  static constexpr bool value = true;
+  using type = std::false_type;
+};
+template <typename Unused>
+struct ReturnTrueFalse {
+  using type = TrueFalse;
+};
+/// [tmpl::count_if:bug:definitions]
+
+void run() noexcept {
+/// [tmpl::all]
+assert_same<tmpl::all<List1<tmpl::size_t<0>, tmpl::size_t<1>, tmpl::size_t<2>>>,
+            tmpl::false_type>();
+assert_same<tmpl::all<List1<tmpl::size_t<1>, tmpl::size_t<1>, tmpl::size_t<2>>>,
+            tmpl::true_type>();
+assert_same<tmpl::all<List1<tmpl::size_t<0>, tmpl::size_t<1>, tmpl::size_t<2>>,
+                      tmpl::less<tmpl::_1, tmpl::size_t<2>>>,
+            tmpl::false_type>();
+assert_same<tmpl::all<List1<tmpl::size_t<0>, tmpl::size_t<1>, tmpl::size_t<0>>,
+                      tmpl::less<tmpl::_1, tmpl::size_t<2>>>,
+            tmpl::true_type>();
+assert_same<tmpl::all<List1<>>, tmpl::true_type>();
+/// [tmpl::all]
+
+/// [tmpl::all:inhomogeneous]
+assert_same<tmpl::all<List1<std::true_type, tmpl::true_type>, tmpl::_1>,
+            tmpl::false_type>();
+/// [tmpl::all:inhomogeneous]
+
+/// [tmpl::any]
+assert_same<tmpl::any<List1<tmpl::size_t<0>, tmpl::size_t<1>, tmpl::size_t<2>>>,
+            tmpl::true_type>();
+assert_same<tmpl::any<List1<tmpl::size_t<0>, tmpl::size_t<0>, tmpl::size_t<0>>>,
+            tmpl::false_type>();
+assert_same<tmpl::any<List1<tmpl::size_t<0>, tmpl::size_t<1>, tmpl::size_t<2>>,
+                      tmpl::less<tmpl::_1, tmpl::size_t<2>>>,
+            tmpl::true_type>();
+assert_same<tmpl::any<List1<tmpl::size_t<4>, tmpl::size_t<3>, tmpl::size_t<2>>,
+                      tmpl::less<tmpl::_1, tmpl::size_t<2>>>,
+            tmpl::false_type>();
+assert_same<tmpl::any<List1<>>, tmpl::false_type>();
+/// [tmpl::any]
+
+/// [tmpl::any:inhomogeneous]
+assert_same<tmpl::any<List1<std::false_type, tmpl::false_type>, tmpl::_1>,
+            tmpl::true_type>();
+/// [tmpl::any:inhomogeneous]
+
+/// [tmpl::at]
+assert_same<tmpl::at<List1<Type1, Type2, Type3>, tmpl::size_t<0>>, Type1>();
+/// [tmpl::at]
+
+/// [tmpl::at_c]
+assert_same<tmpl::at_c<List1<Type1, Type2, Type3>, 0>, Type1>();
+/// [tmpl::at_c]
+
+/// [tmpl::back]
+assert_same<tmpl::back<List1<Type1, Type2, Type3>>, Type3>();
+/// [tmpl::back]
+
+/// [tmpl::count_if]
+assert_same<tmpl::count_if<List1<Type1, Type2, Type1>,
+                           std::is_same<tmpl::_1, Type1>>,
+            tmpl::integral_constant<size_t, 2>>();
+/// [tmpl::count_if]
+
+/// [tmpl::count_if:bug:asserts]
+assert_same<tmpl::count_if<List1<Type1>, ReturnTrueFalse<void>>,
+            tmpl::integral_constant<size_t, 0>>();
```

Comment:
Add comment // should return tmpl::integral_constant<size_t, 1>

Reply:
The documented bugs have been fixed, so these sections will be removed.